### ISSUE TITLE
1.3.1 breaks rake migrations

### ISSUE THREAD

**Issue by archonic** [Dynamoid/Dynamoid#139]
Using 1.3.1 I'm able to load the application (start a console, query things), but when running `rake db:migrate`, it errors out with `NameError: uninitialized constant Account::Exportable`. The error is non-sense, it's just complaining about including the first module it finds in the first file it checks. Moving back to dynamoid 1.1.0 fixes it. That's a very serious issue, obviously.

**Comment by pboling** [comment]
Serious indeed.  I will investigate this.

**Comment by ruslankhaertdinov** [comment]
Same issue: `NameError: uninitialized constant Angle::LatitudeValidator` and also with `rake db:migrate` and `1.3.1`

**Comment by pboling** [comment]
Was introduced between 79dab0c7ea49480a40c021bfe632f131b237cbcc..eb3965d6920466894192bbcf49188afffc585be3

**Comment by nolantait** [comment]
I think it's actually because of the way models are loaded. At least it was for my specific problem.

Line 6 of https://github.com/Dynamoid/Dynamoid/blob/master/lib/dynamoid/tasks/database.rake
`Dir[ File.join(MODELS, "*.rb") ].sort.each { |file| require file }`

Removing that line allows it to skip trying to require models out of order. There is probably a better way to load models from a rails app that may depend on other files within the application.

**Comment by pboling** [comment]
Wow.  @nolantait - you are right.

The problem is straightforward.  You can't load Rails models outside rails very well.  Rails is not designed to require dependencies in every file, rather it is designed to auto-load things.  But outside the rails environment the auto load paths are not configured, so most things won't be able to be found by Ruby.

The `=> :environment` option makes the task load the rails env, but then it is only loaded inside the task.

```
  task :create_tables => :environment do
```

So the loading of models - if it is needed at all - should only happen inside the task definition.

Fixing.

**Comment by pboling** [comment]
Released 1.3.2 with the fix.

```
  task :create_tables => :environment do
    # Load models so Dynamoid will be able to discover tables expected.
    Dir[ File.join(MODELS, "*.rb") ].sort.each { |file| require file }
    tables = Dynamoid::Tasks::Database.create_tables
    result = tables[:created].map{ |c| "#{c} created" } + tables[:existing].map{ |e| "#{e} already exists" }
    result.sort.each{ |r| puts r }
  end
```
